tests/deployment/vlan_networks.py

```diff
@@ -173,6 +216,15 @@ def delete_networks():
                     all_attachments.append((node.label,
                                             node.nics[0].label,
                                             attachment.network.label))
+
+            switch = nodes[0].nics[0].port.owner
+            # in some switches, the native network can only be disconnected
+            # after we remove all tagged networks first. The following checks
+            # for that and rearranges the networks (all_attachments) such that
+            # tagged networks are removed first.
+            if 'nativeless-trunk-mode' not in switch.get_capabilities():
+                all_attachments = sorted(all_attachments)
```

Comment:
Since you've run the tests I imagine this *does* work, but it would otherwise take a bit of staring at the code for me to be sure that sorting this actually results in the tagged ones first. Can you do this in a way that's more self-explanatory (even if it's a bit more verbose)?